text/1521-copy-clone-semantics.md

```diff
@@ -34,15 +34,19 @@ compatible to upgrade to Clone in the future if demand is high enough."
 # Detailed design
 [design]: #detailed-design
 
-Specify that `<T as Clone>::clone(t)` shall be equivalent to `ptr::read(t)`
-where `T: Copy, t: &T`. An implementation that does not uphold this *shall not*
-result in undefined behavior; `Clone` is not an `unsafe trait`.
+Specify that users may assume `<T as Clone>::clone(t)` to be equivalent to
+`ptr::read(t)` where `T: Copy, t: &T`. An implementation that does not uphold
+this *shall not* result in undefined behavior; `Clone` is not an `unsafe
+trait`.
 
 Also add something like the following sentence to the documentation for the
 `Clone` trait:
 
 "If `T: Copy`, `x: T`, and `y: &T`, then `let x = y.clone();` is equivalent to
-`let x = *y;`. Manual implementations must be careful to uphold this."
+`let x = *y;`. Manual implementations should be careful to uphold this
+invariant, as users *may* assume that it holds (however the compiler itself
+*will not* make that assumption when compiling calls to `Clone::clone`,
+guaranteeing that the relevant `clone` method is always invoked)."
```

Comment:
I think this might be easier to understand by drawing a parallel to `PartialEq`: we expect implementations to uphold a certain contract, *but nobody may rely on that for the purpose of `unsafe` preconditions / memory safety*. This idea has been around since Rust 1.0, it is well understood, so we should build on it.